File: k-kbk/1주차/2798.py
print(result)

from itertools import combinations

N, M = map(int, input().split())
cards = list(map(int, input().split()))

result = 0

for combination in combinations(cards, 3):
    total = sum(combination)
    if result < total <= M:
        result = total

print(result)

# - 풀이 2

# N, M = map(int, input().split(" "))
# cards = list(map(int, input().split(" ")))

# result = 0

# for i in range(N):
#     for j in range(N):
#         for k in range(N):
#             if i == j or j == k or k == i:
#                 continue

#             number = cards[i] + cards[j] + cards[k]
#             if result < number <= M:
#                 result = number

File: keemsebin/1주차/2798.js
const readline = require("readline");
const rl = readline.createInterface({
  input: process.stdin,
  output: process.stdout,
});

let input = [];

rl.on("line", function (line) {
  input.push(line);
}).on("close", function () {
  let [N, M] = input[0].split(" ").map(Number);
  let arr = input[1].split(" ").map(Number);
  let result = 0;

  for (let i = 0; i < N - 2; i++) {
    for (let j = i + 1; j < N - 1; j++) {
      for (let k = j + 1; k < N; k++) {
        let sum = arr[i] + arr[j] + arr[k];
        if (sum <= M && sum > result) {
          result = sum;
        }
      }
    }
  }
  console.log(result);
});
